tools/port-drayage-webservice/src/main/resources/port-drayage-webservice.yml

```diff
@@ -0,0 +1,384 @@
+      responses:
+        '201':
+          description: Created
+        '400':
+          description: Bad Request
+  /inspection/pending:
+    get:
+      summary: List of all pending inspection actions and status.
+      description: Get all inspection actions that have been requested by are not yet in progress.
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/InspectionStatusList'  
+  /inspection/complete/{action_id}:
+    post:
+      parameters:
+        - in: path
+          name: action_id
+          schema:
+            type: string
+          required: true
+          description: ID of action
+      summary: Complete an inspection action.
+      description: Will attempt to complete current inspection action. Will return 400 if no current inspection
+      responses:
+        '201':
+          description: Created   
+        '400':
+          description: Bad Request
+  /inspection/hold/{action_id}:
+    post:
+      parameters:
+        - in: path
+          name: action_id
+          schema:
+            type: string
+          required: true
+          description: ID of action
+      summary: Request vehicle proceed to holding area for further inspection
+      description: Will attempt to request vehicle proceed to holding area for further inspection. Will return 400 if no current action
+      responses:
+        '201':
+          description: Created   
+        '400':
+          description: Bad Request   
+  /inspection/{action_id}:
+    get:
+      parameters:
+        - in: path
+          name: action_id
+          schema:
+            type: string
+          required: true
+          description: ID of action
+      summary: Returns action with given ID 
+      description: Returns action with given ID. If non is found returns 400
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/InspectionStatus'
+        '400':
+          description: Bad Request    
+components:
+  schemas:
+    ContainerRequest:
+      title: Container Request
+      type: object
+      properties:
+        vehicle_id:
+          type: string
+          description: ID of vehicle
+        container_id:
+          type: string
+          description: ID of container
+        action_id:
+          type: string
+          description: ID of action
+      required:
+        - vehicle_id
+        - container_id
+        - action_id
+    ContainerActionStatus:
+      title: Container Action Status
+      type: object
+      properties:
+        vehicle_id:
+          type: string
+          description: ID of vehicle
+        container_id:
+          type: string
+          description: ID of container
+        action_id:
+          type: string
+          description: ID of action

```

Comment:
Why do we need an action status? and can we provide duplicate action for the same vehicle and container?

Reply:
Do you mean action_id. We need action_id for being able to uniquely identify each action. We use it in the PortDrayage MySQL database to reference each action. This way I no longer have to check vehicle and container id to look for a specific action.  I think this is a good way to index the actions. 

We need action_status to monitor the status of a given action. As a container is being loaded we need a way to describe its progress in this process to be able to notify V2X-Hub Port Drayage plugin that it has been completed.

Reply:
The action id is from client input. Does that mean the client is responsible for generating unique action id for every request?

Reply:
What is the list of the action status?

Reply:
There is a request id that seems to be unique for every request, so action id and request id are different?